torch/_functorch/aot_autograd.py

```diff
@@ -1869,12 +1869,28 @@ class CompiledFunctionBackward(torch.autograd.Function):
                 def forward(ctx, *all_args):
                     all_args_list = list(all_args)
                     if CompiledFunction.compiled_bw is None:
-                        # TODO - pass in fake tensors ?
                         context = disable_autocast_manager if disable_amp else nullcontext
-                        with context(), track_graph_compiling(aot_config, "backward"):
-                            CompiledFunction.compiled_bw = aot_config.bw_compiler(
-                                bw_module, all_args_list
-                            )
+                        if config.use_dynamic_shapes:
+                            with context():
+                                # Retrace backwards with a new shape environment
+                                local_bw_module = make_fx(bw_module, tracing_mode="symbolic")(*all_args)
+                                with track_graph_compiling(aot_config, "backward"):
+                                    CompiledFunction.compiled_bw = aot_config.bw_compiler(
+                                        # NB: The None case is for when we
+                                        # have SymInt tangents, which are
+                                        # always passed with None
```

Comment:
dumb q: that comment seems to imply that the "SymInt tangent" case lines up with when `'val' not in n.meta`. But I thought that proxy nodes representing SymInts **do** get their `n.meta['val']` populated. Only "constant" nodes (plain numeric types/strings/output node/etc don't have that field)

Reply:
It's not the SymInt input that is a problem (that indeed has a meta), but the *tangent* of that input, which is None and doesn't have a meta.

TBH though these Nones are tiresome and I kind of would like it if we could change the calling convention so SymInt inputs don't get tangents